src/robots/Robot.java

```diff
@@ -0,0 +1,39 @@
+package robots;
+
+public class Robot implements Shoot, Move, Yell, Defence {
+    private String name;
+    private String version;
```

Comment:
you craeted version field, but it is string. well, it's ok, but I want you to use Enum. please read/watch about enums in java and change this field from String to Enum and provide different enum for each version.

Reply:
added enum, used in abstract class